src/EprPrnIntegration.Common/Service/ServiceBusProvider.cs

```diff
@@ -1,45 +1,126 @@
 ﻿using Azure.Messaging.ServiceBus;
 using EprPrnIntegration.Common.Configuration;
 using EprPrnIntegration.Common.Models;
+using EprPrnIntegration.Common.Models.Queues;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using System.Text.Json;
 
 namespace EprPrnIntegration.Common.Service
 {
-    public class ServiceBusProvider : IServiceBusProvider
+    public class ServiceBusProvider(
+        ILogger<ServiceBusProvider> logger,
+        ServiceBusClient serviceBusClient,
+        IOptions<ServiceBusConfiguration> config) : IServiceBusProvider
     {
-        private readonly ILogger<ServiceBusProvider> _logger;
-        private readonly ServiceBusClient _serviceBusClient;
-        private readonly IOptions<ServiceBusConfiguration> _serviceBusConfig;
-        public ServiceBusProvider(ILogger<ServiceBusProvider> logger, ServiceBusClient serviceBusClient, IOptions<ServiceBusConfiguration> config)
-        {
-            _logger = logger;
-            _serviceBusClient = serviceBusClient;
-            _serviceBusConfig = config;
-        }
         public async Task SendFetchedNpwdPrnsToQueue(List<NpwdPrn> prns)
         {
             try
             {
-                await using var sender = _serviceBusClient.CreateSender(_serviceBusConfig.Value.FetchPrnQueueName);
+                await using var sender = serviceBusClient.CreateSender(config.Value.FetchPrnQueueName);
                 using ServiceBusMessageBatch messageBatch = await sender.CreateMessageBatchAsync();
                 foreach (var prn in prns)
                 {
                     var jsonPrn = JsonSerializer.Serialize(prn);
                     if (!messageBatch.TryAddMessage(new ServiceBusMessage(jsonPrn)))
                     {
-                        _logger.LogWarning("SendFetchedNpwdPrnsToQueue - The message with EvidenNo: {EvidenNo} is too large to fit in the batch.", prn.EvidenceNo);
+                        logger.LogWarning("SendFetchedNpwdPrnsToQueue - The message with EvidenNo: {EvidenNo} is too large to fit in the batch.", prn.EvidenceNo);
```

Comment:
I don't know the usage so its not clear to me whether {EvidenNo} is correct or it should be {EvidenceNo}. If it should be {EvidenceNo} please change

Reply:
I reckon this is referring to Eviden as the company number. not sure tbh as its an existing code and I only changed the logger and serviceBusClient variable name

Reply:
We can ignore this it should be evidenceNo but this is going to change with upcoming ticket which venkata is working